### PR TITLE
[go] implement RASP endpoints

### PR DESCRIPTION
**Pull request by eliottness**
## Motivation

Support RASP tests

## Changes

Implement rasp endpoints as per spec in `docs/weblog/README.md`

## Workflow

1. ⚠️ Create your PR as draft ⚠️
2. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
3. Mark it as ready for review
    * Test logic is modified? -> Get a review from RFC owner. We're working on refining the `codeowners` file quickly.
    * Framework is modified, or non obvious usage of it -> get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

:rocket: Once your PR is reviewed, you can merge it!

🛟 [#apm-shared-testing](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X) 🛟

## Reviewer checklist

* [x] [Relevant label](https://github.com/DataDog/system-tests/blob/main/docs/CI/labels.md) (`run-parametric-scenario`, `run-profiling-scenario`...) are presents
* [x] If PR title starts with `[<language>]`, double-check that only `<language>` is impacted by the change
* [x] No system-tests internal is modified. Otherwise, I have the approval from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
* [x] CI is green, or failing jobs are not related to this change (and you are 100% sure about this statement)
* [ ] A docker base image is modified?
    * [ ] the relevant `build-XXX-image` label is present
* [ ] A scenario is added (or removed)?
    * [ ] Get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

### CODE REVIEW

File: utils/build/docker/golang/app/internal/rasp/rasp.go [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+		return
+	}
+
+	defer db.Close()
+
+	if _, err = db.Exec("CREATE TABLE users (id INTEGER PRIMARY KEY, name TEXT)"); err != nil {
+		log.Fatalln(err.Error())
+	}
+
+	_, err = db.Exec("SELECT * FROM users WHERE name = '" + sqli)
```
Comment (eliottness) [inline utils/build/docker/golang/app/internal/rasp/rasp.go:123]:
```suggestion
	_, err = db.Exec("SELECT * FROM users WHERE name = '" +sqli+"'")
```

File: utils/build/docker/golang/app/internal/rasp/rasp.go [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+			log.Fatalf("failed to parse body: %v\n", err)
+		}
+
+		if _, ok := body[key]; !ok {
+			log.Fatalln("missing key in body: ", key)
+		}
+
+		if err := appsec.MonitorParsedHTTPBody(r.Context(), body); err != nil {
+			log.Fatalf("Body Monitoring should not block the request: %v\n", err)
+		}
```
Comment (Hellzy) [inline utils/build/docker/golang/app/internal/rasp/rasp.go:51]: Is this really necessary? Can't we leave that to the actual body monitoring tests?

File: utils/build/docker/golang/app/internal/rasp/rasp.go [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+}
+
+func SQLi(w http.ResponseWriter, r *http.Request) {
+	sqli := parseRASPRequest(r, "user_id")
+	if sqli == "" {
+		return
+	}
+
+	sql.Register("sqlite3", &sqlite3.SQLiteDriver{})
+	db, err := sql.Open("sqlite3", ":memory:")
```
Comment (Hellzy) [inline utils/build/docker/golang/app/internal/rasp/rasp.go:110]: ``
```suggestion
        // TODO: use sqltrace pkg once RASP SQLi gets merged in dd-trace-go
	db, err := sql.Open("sqlite3", ":memory:")
```

File: utils/build/docker/golang/app/internal/rasp/rasp.go [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+	return input
+}
+
+func LFI(w http.ResponseWriter, r *http.Request) {
+	path := parseRASPRequest(r, "file")
+	if path == "" {
+		return
+	}
+
+	_, err := os.ReadFile(path)
```
Comment (Hellzy) [inline utils/build/docker/golang/app/internal/rasp/rasp.go:70]:
```suggestion
        // TODO: use lfitrace (or equivalent) pkg once RASP LFI is implemented and merged in dd-trace-go
	_, err := os.ReadFile(path)
```

**Review by Hellzy — COMMENTED** [review]
Thanks for this.
Is the go.work.sum supposed to be modified?

**Review by robertomonteromiguel — CHANGES_REQUESTED** [review]
should we activate the tests related to these endpoint with the manifest file?

**Comment by eliottness** [comment]
> should we activate the tests related to these endpoint with the manifest file?

@robertomonteromiguel We did and it uncovered a bug so we are doing a patch release 😅

 All 3 endpoints have been tested tho

**Comment by eliottness** [comment]
> Thanks for this.
> 
> Is the go.work.sum supposed to be modified? 

@Hellzy I guess so because it gets modified along the go.mod file when I upgrade dd-trace-go

**Review by robertomonteromiguel — APPROVED** [review]

**Pull request merged**